Add `proxyPayloads` option (#87)
fastify/fastify-http-proxy: README.md

A `preHandler` to be applied on all routes. Useful for performing actions before the proxy is executed (e.g. check for authentication).

### proxyPayloads

When you set this option as false, you will be able to access the body but it would also disable the direct pass through of the payload. The implementor will have to take care if the payload is proxied correctly.

> 

Reviewer: Could you add an example of usage of this, as it's easy to mess up.

Author: yeah will do that

Author: Hey,I was thinking... will it be an example like just adding a contentParser for some content-type on fastify or a method actually parsing some data of some content-type?

Reviewer: Whatever you want.

> The default value of this option is true. The ContentTypeParsers are removed by default for proxy but incase you want to access the body to take certain decision like authentication or authorization then you can set this option to false.

Reviewer 2: ```suggestion
Setting this option to `false` will disable pass through of proxied payloads. When setting this to `false`, it is left up to the implementor to handle proxied payloads correctly.
```

Reviewer 2: There should also be an example here showing what handling of a payload looks like.

Author: Hi @jsumners,
I don't think that making this option false will disable pass through of proxied payloads. It would even then pass the payload to the upstream.

If you are thinking of some ContentType which would create problems in parsing then please do give example because I might not know about it..

A `preHandler` to be applied on all routes. Useful for performing actions before the proxy is executed (e.g. check for authentication).

### proxyPayloads

When you set this option as `false`, you will be able to access the body but it would also disable the direct pass through of the payload. The implementor will have to take care if the payload is parsed and proxied correctly.

For Example,

If you are expecting payload of type application/xml, then you would have to add a parser for it like this.

```javascript

> fastify.addContentTypeParser('application/xml', (req, reply) => {
>   //parsing code

Reviewer 3: ```suggestion
fastify.addContentTypeParser('application/xml', (req, done) => {
  const parsedBody = parsingCode(req)
  done(null, parsedBody)
```

### proxyPayloads

> When you set this option as `false`, you will be able to access the body but it would also disable the direct pass through of the payload. The implementor will have to take care if the payload is parsed and proxied correctly.
> 
> For Example,
> 
> If you are expecting payload of type application/xml, then you would have to add a parser for it like this.
> 

Reviewer 2: ```suggestion
When this option is `false`, you will be able to access the body but it will also disable direct pass through of the payload. As a result, it is left up to the implementation to properly parse and proxy the payload correctly.

For example, if you are expecting a payload of type `application/xml`, then you would have to add a parser for it like so:

```
